gammapy/background/reflected.py

```diff
@@ -198,36 +181,58 @@ def setup(self):
             self._angle + Angle("360deg") - self._min_ang - self.min_distance_input
         )
 
-        # Distance image
-        self._distance_image = _compute_distance_image(self.reference_map)
+        if self._min_ang < 0:
+            print("ISSUE self._min_ang=", self._min_ang)
+        if self.min_distance_input < 0:
+            print("ISSUE self.min_distance_input=", self.min_distance_input)
 
     def find_regions(self):
         """Find reflected regions."""
         curr_angle = self._angle + self._min_ang + self.min_distance_input
         reflected_regions = []
+        geom = self.reference_map.geom
+        _run_exclusion_mask = None
+        _pixel_excluded = False
+        if not self.exclusion_mask is None:
+            _run_exclusion_mask = self.exclusion_mask.reproject(self.reference_map.geom)
+            _run_exclusion_mask.data[np.where(np.isnan(_run_exclusion_mask.data))] = 1
+            _mask_array = np.where(_run_exclusion_mask.data < 0.99)
+            _excluded_coords = geom.get_coord().apply_mask(_mask_array).to_coordsys('CEL')
+            if len(_excluded_coords[0]) > 0 and len(_excluded_coords[1]) > 0 :
+                _excluded_pixcoords = PixCoord(*SkyCoord(*_excluded_coords, unit='deg').to_pixel(geom.wcs))
+                _pixel_excluded = True
+
         while curr_angle < self._max_angle:
-            test_pos = self._compute_xy(self._pix_center, self._offset, curr_angle)
-            test_reg = CirclePixelRegion(test_pos, self._pix_region.radius)
-            if not self._is_inside_exclusion(test_reg, self._distance_image):
-                refl_region = test_reg.to_sky(self.reference_map.geom.wcs)
-                log.debug("Placing reflected region\n{}".format(refl_region))
-                reflected_regions.append(refl_region)
+            _test_reg = self._create_rotated_reg(curr_angle)
+            if _pixel_excluded is False or not np.any(_test_reg.contains(_excluded_pixcoords)):
+                _region = _test_reg.to_sky(geom.wcs)
+                log.debug("Placing reflected region\n{}".format(_region))
+                reflected_regions.append(_region)
                 curr_angle = curr_angle + self._min_ang
                 if self.max_region_number <= len(reflected_regions):
                     break
             else:
                 curr_angle = curr_angle + self.angle_increment
 
+        print("Found {0} reflected regions for the Obs #{1}".format(len(reflected_regions), self.obs_id))
         log.debug("Found {} reflected regions".format(len(reflected_regions)))
+
         self.reflected_regions = reflected_regions
 
+        # Make the OFF reference map
+        _mask = geom.region_mask(self.reflected_regions, inside=True)
+        self.off_reference_map = WcsNDMap(geom=geom, data=_mask)
+        # TODO: remove this lign after the correction of the Issue #2074
+        self.off_reference_map.data = self.off_reference_map.data.astype(float)
```

Comment:
Would a `SkyRegion` based event selection make things easier regarding bin size?

Reply:
This functionality was not present at the time of the PR. If it exists, one could use also the proposed solution